weave

```diff
@@ -74,9 +74,15 @@ EOF
 }
 
 exec_remote() {
+    HOSTS_PATH=$(docker $DOCKER_CLIENT_ARGS run --rm --net=none \
+        -v /var/run/docker.sock:/var/run/docker.sock \
+        --entrypoint /bin/sh \
+        $EXEC_IMAGE -c 'docker inspect -f {{.HostsPath}} $(hostname)')
```

Comment:
That's sort-of what I meant, but I just realised this is a change too far for this PR, since we are not going to do the edit for `weave run` or `weave attach`.  The reason to defer is it then seems necessary to unwind it in `weave detach`.